File: 3.LongestNonRepeatingSubstring/Solution.java
import java.util.HashSet;
import java.util.Set;

public class Solution {
    public int longestSubstring(String s) {
        int i = 0, j = 0, ans = 0, n = s.length();
        Set<String> ht = new HashSet<String>();
        while (j < n) {
            while(ht.contains(s.substring(i, 1)))
                ht.remove(s.substring(i++, 1));
            ans = Math.max(ans, j - i + 1);
            ht.add(s.substring(j, 1));
        }
        return ans;
    }
}

File: 3.LongestNonRepeatingSubstring/Solution.py
class Solution:
    def lengthOfLongestSubstring(self, s:str):
        mp = [False] * 256
        i,j = 0 , 0
        n = len(s)
        ans = 0
        while j < n:
            while(mp[ord(s[j])]):
                mp[ord(s[i])] = False
                i += 1
            mp[ord(s[j])] = True
            ans = max(ans, j - i + 1)
            j += 1
        return ans